cpmpy/solvers/ortools.py

```diff
@@ -472,17 +473,30 @@ def _post_constraint(self, cpm_expr, reifiable=False):
                 return self.ort_model.AddCumulative(intervals, demand, cap)
             elif cpm_expr.name == "circuit":
                 # ortools has a constraint over the arcs, so we need to create these
-                # when using an objective over arcs, using these vars direclty is recommended
+                # when using an objective over arcs, using these vars directly is recommended
                 # (see PCTSP-path model in the future)
                 x = cpm_expr.args
                 N = len(x)
                 arcvars = boolvar(shape=(N,N))
                 # post channeling constraints from int to bool
                 self += [b == (x[i] == j) for (i,j),b in np.ndenumerate(arcvars)]
                 # post the global constraint
-                # when posting arcs on diagonal (i==j), it would do subcircuit
+                # when posting arcs on diagonal (i==j), it would do subcircuit (see subcircuit)
                 ort_arcs = [(i,j,self.solver_var(b)) for (i,j),b in np.ndenumerate(arcvars) if i != j]
                 return self.ort_model.AddCircuit(ort_arcs)
+            elif cpm_expr.name == "subcircuit":
+                x = cpm_expr.args
+                N = len(x)
+                arcvars = boolvar(shape=(N,N))
+                # post channeling constraints from int to bool
+                self += [b == (x[i] == j) for (i,j),b in np.ndenumerate(arcvars)]
+                # post the global constraint
+                # posting arcs on diagonal (i==j) allows for subcircuits
+                ort_arcs = [(i,j,self.solver_var(b)) for (i,j),b in np.ndenumerate(arcvars) if not ((i == j) and (i == cpm_expr.startIndex))] # The start index cannot self loop and thus must be part of the subcircuit.
+                # If no startIndex is given, garuantee that subcircuit has at least length 2
+                if cpm_expr.startIndex is None:
+                    self += [cpm_any(cpm_array(x) != np.arange(N))]
```

Comment:
Here it does not matter the restriction is put to toplevel as OR-tools does not support reified subcircuit anyway, so fine to keep it here BUT needs to change in Choco